src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java

```diff
@@ -0,0 +1,159 @@
+        this.mlClient = mlClient;
+    }
+
+    @Override
+    public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) throws IOException {
+        CompletableFuture<WorkflowData> createConnectorFuture = new CompletableFuture<>();
+
+        ActionListener<MLCreateConnectorResponse> actionListener = new ActionListener<>() {
+
+            @Override
+            public void onResponse(MLCreateConnectorResponse mlCreateConnectorResponse) {
+                logger.info("Created connector successfully");
+                // TODO Add the response to Global Context
+                createConnectorFuture.complete(
+                    new WorkflowData(Map.ofEntries(Map.entry("connector_id", mlCreateConnectorResponse.getConnectorId())))
+                );
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error("Failed to create connector");
+                createConnectorFuture.completeExceptionally(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+            }
+        };
+
+        String name = null;
+        String description = null;
+        String version = null;
+        String protocol = null;
+        Map<String, String> parameters = new HashMap<>();
+        Map<String, String> credentials = new HashMap<>();
+        List<ConnectorAction> actions = null;
+
+        for (WorkflowData workflowData : data) {
+            Map<String, Object> content = workflowData.getContent();
+
+            for (Entry<String, Object> entry : content.entrySet()) {
+                switch (entry.getKey()) {
+                    case NAME_FIELD:
+                        name = (String) content.get(NAME_FIELD);
+                        break;
+                    case DESCRIPTION:
+                        description = (String) content.get(DESCRIPTION);
+                        break;
+                    case VERSION_FIELD:
+                        version = (String) content.get(VERSION_FIELD);
+                        break;
+                    case PROTOCOL_FIELD:
+                        protocol = (String) content.get(PROTOCOL_FIELD);
+                        break;
+                    case PARAMETERS_FIELD:
+                        parameters = getParameterMap((Map<String, String>) content.get(PARAMETERS_FIELD));
+                        break;
+                    case CREDENTIALS_FIELD:
+                        credentials = (Map<String, String>) content.get(CREDENTIALS_FIELD);
+                        break;
+                    case ACTIONS_FIELD:
+                        actions = (List<ConnectorAction>) content.get(ACTIONS_FIELD);
+                        break;
+                }
+
+            }
+        }
+
+        if (Stream.of(name, description, version, protocol, parameters, credentials, actions).allMatch(x -> x != null)) {
+            MLCreateConnectorInput mlInput = MLCreateConnectorInput.builder()
```

Comment:
We should complete exceptionally if any of these fields are null. The action listener wont be hit if we dont end up executing the create connector request